File: EmptyProject/WorkflowsTrainingTwo/WorkflowsTraining/Parsing/Expressions/Nodes/Expressions/IdentifierNode.cs
using WorkflowsTraining.Helpers.Storage;
using WorkflowsTraining.Parsing.Expressions.Interfaces;

namespace WorkflowsTraining.Parsing.Expressions.Nodes.Expressions;

public class IdentifierNode(string name) : IExpressionNode
{
    public string Name { get; } = name;

    public object Resolve()
    {
        return VariableStorage.GetVariable(Name);
    }
}

File: EmptyProject/WorkflowsTrainingTwo/WorkflowsTrainingCPlus/Parsing/Expressions/Nodes/Expressions/IdentifierNode.cpp
#include "IdentifierNode.h"
#include <iostream>

namespace WorkflowsTraining {
    namespace Parsing {
        namespace Expressions {
            namespace Nodes {
                namespace Expressions {
                    IdentifierNode::IdentifierNode(std::string name)
                        : _name(std::move(name)) {
                    }

                    std::string IdentifierNode::GetName() const
                    {
                        return _name;
                    }

                    std::variant<bool, int, double, std::string> IdentifierNode::Resolve() {
                        std::cout << "IdentifierNode: Resolving '" << _name << "'..." << std::endl;
                        //return WorkflowsTraining::Helpers::Storage::VariableStorage::GetVariable(_name);
                    }
                }
            }
        }
    }
}